unit-tests/syncer/test-ts-eof.py

```diff
@@ -0,0 +1,150 @@
+#
+test.start( "Init" )
+
+# It can take a few frames for the syncer to actually produce a matched frameset (it doesn't
+# know what to match to in the beginning)
+
+sw.generate_depth_and_color( frame_number = 0, timestamp = 0 )
+sw.expect( depth_frame = 0 )                          # syncer doesn't know about color yet
+sw.expect( color_frame = 0, nothing_else = True )     # less than next expected of D
+#
+# NOTE: if the syncer queue wasn't 100 (see above) then we'd only get the color frame!
+#
+sw.generate_depth_and_color( 1, sw.gap_d * 1 )
+sw.expect( depth_frame = 1, color_frame = 1, nothing_else = True )  # frameset 1
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Keep going" )
+
+sw.generate_depth_and_color( 2, sw.gap_d * 2 )
+sw.expect( depth_frame = 2, color_frame = 2, nothing_else = True )  # frameset 2
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Stop giving color; nothing output" )
+
+sw.generate_depth_frame( 3, sw.gap_d * 3 )
+
+# The depth frame will be kept in the syncer, and never make it out (no matching color frame
+# and we're not going to push additional frames that would cause it to eventually flush):
+#
+sw.expect_nothing()
+#
+# ... BUT the file should still have it!
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Dump the file" )
+
+recorder.pause()
+recorder = None  # otherwise the file will be open when we exit
+log.d( "filename=", filename )
+sw.stop()
+sw.reset()
+#
+# Dump it... should look like:
+#     [Depth/0 #0 @0.000000]
+#     [Color/1 #0 @0.000000]
+#     [Depth/0 #1 @16.666667]
+#     [Color/1 #1 @16.666667]
+#     [Depth/0 #2 @33.333333]
+#     [Color/1 #2 @33.333333]
+#     [Depth/0 #3 @50.000000]    <--- the frame that was "lost"
+#
+import subprocess, sys
+for p in sys.path:
+    rs_convert = os.path.join( p, 'rs-convert.exe' )
+    if os.path.isfile( rs_convert ):
+        subprocess.run( [rs_convert, '-i', filename, '-T'],
+                        stdout=None,
+                        stderr=subprocess.STDOUT,
+                        universal_newlines=True,
+                        timeout=10,
+                        check=True )
+        break
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Play it back, with syncer -- lose last frame" )
+
+sw.playback( filename )
+sw.start()
+
+sw.expect( depth_frame = 0 )                          # syncer doesn't know about color yet
+sw.expect( color_frame = 0 )                          # less than next expected of D
+sw.expect( depth_frame = 1, color_frame = 1 )
+sw.expect( depth_frame = 2, color_frame = 2 )
+
+# We know there should be another frame in the file:
+#     [Depth/0 #3 @50.000000]
```

Comment:
Why @50 and not @60?

Reply:
See the dump -- frame #3 was generated at `gap_d*3` so `16.67*3` or ~`50`

Reply:
You are right, I confused reading time as FPS.